app/models/aleph_hint.rb

```diff
@@ -0,0 +1,62 @@
+# Parses Aleph marcxml and creates Hints
+#
+require 'open-uri'
+class AlephHint
+  def initialize
+    @marcxml = marcxml
+  end
+
+  def marcxml
+    Nokogiri::XML(open(Rails.application.secrets.aleph_hint_source))
+  end
+
+  def records
+    @marcxml.xpath('//xmlns:record')
+  end
+
+  def process_records
+    records.each do |record|
+      record_to_hints(record)
+    end
+  end
+
+  # we need to identify all fingerprints and then create a hint for each one
+  # fingerprints will be created from each title and alt title
+  def record_to_hints(record)
+    Hint.create(title: title(record), url: url(record),
+                fingerprint: Hint.fingerprint(title(record)))
+
+    fingerprint_alt_titles(246, record)
+
+    fingerprint_alt_titles(740, record)
+  end
+
+  def fingerprint_alt_titles(field, record)
+    alt_title(field, record).each do |hint|
+      Hint.create(title: title(record), url: url(record),
+                  fingerprint: Hint.fingerprint(hint))
+    end
+  end
+
+  def title(record)
+    extract_marc_value(245, 'a', record).text
+  end
+
+  def alt_title(tag, record)
+    extract_marc_value(tag, 'a', record).map(&:text)
+  end
+
+  def extract_marc_value(tag, subfield, record)
+    record.xpath(
+      "./xmlns:datafield[@tag=#{tag}]/xmlns:subfield[@code=\"#{subfield}\"]"
+    )
+  end
+
+  def url(record)
+    if extract_marc_value(856, 'u', record).present?
+      extract_marc_value(856, 'u', record).text
+    else
+      extract_marc_value(956, 'u', record).text
```

Comment:
Unless the datasource changes, which would involve a discussion we'd be involved in, we can assume all records have a URL. If that contract is broken, we throw exceptions because things are broken.